scripts/build_shapes.py

```diff
@@ -836,4 +892,10 @@ def gadm(
         year,
         nprocesses=nprocesses,
     )
+    gadm_shapes.geometry = gadm_shapes.apply(
```

Comment:
In the main some checks are already added; see: https://github.com/pypsa-meets-earth/pypsa-earth/blob/b745abe78090b634c0840bf9c289c35eb51e05ca/scripts/build_shapes.py#L787

It may be better to review this according to the new main.
This code may be moved there: the make_valid is not done but it should as you did.
Maybe apply is not the best function to be used on the entire dataframe.
We could do that directly on the series.
Something like this may do: `gadm_shapes.geometry.apply(lambda r: make_valid(r) if not r.is_valid else r)`?
However, I'd like this to be merged into the gadm function rather than here.

Reply:
Revised